cirq/circuits/circuit_test.py

```diff
@@ -1252,7 +1252,7 @@ def test_findall_operations(circuit_cls):
     zb = cirq.Z.on(b)
 
     def is_x(op: cirq.Operation) -> bool:
-        return isinstance(op, cirq.GateOperation) and isinstance(op.gate, cirq.XPowGate)
+        return op.gate is not None and isinstance(op.gate, cirq.XPowGate)
```

Comment:
Can remove the first check since `None` is not an instance of `cirq.XPowGate`:
```suggestion
        return isinstance(op.gate, cirq.XPowGate)
```
This applies generally throughout this PR; if we're just checking whether `op.gate` is an instance of some class, there's no need to first check whether it is `None`.